lib/pierre.py

```diff
@@ -150,14 +153,20 @@ def get_bodies(event_object):
     return bodies
 
 
-def get_dependencies_from_bodies(bodies):
+def get_dependencies_from_bodies(bodies, dependency_id):
     dependencies = []
     for body in bodies:
         deps = extract_dependency_id(body)
         if deps:
             dependencies.extend(deps)
 
-    return list(set(dependencies))
+    unique_dependencies = set(dependencies)
```

Comment:
you can avoid this new with a simple addition in the `if` above
`if deps && deps != dependency_id:`